processing/src/main/java/io/druid/query/topn/DimExtractionTopNAlgorithm.java

```diff
@@ -139,7 +116,7 @@ protected void updateResults(
         }
 
         resultBuilder.addEntry(
-            entry.getKey(),
+            entry.getKey() == null ? null : entry.getKey().toString(),
```

Comment:
the key can be null for null row values, e.g. testTopNWithNullProducingDimExtractionFn() in TopNQueryRunnerTest